### ISSUE TITLE
"Unexpected modifier sealed" error when using sealed classes

### ISSUE THREAD

**Issue by maxandersen** [openrewrite/rewrite#2738]
we got a bug report at https://github.com/quarkusio/quarkus/issues/30639 for a user trying to do the Quarkus 3 migration steps that uses openrewrite.

It fails because openrewrite fails at parsing classes with "sealed" modifier.

The error we get is:
```
[WARNING] Failed to convert for the following cursor stack:--- BEGIN PATH ---
JCCompilationUnit(sourceFile = /Users/manderse/temp/quarkus-timed-producers/src/main/java/org/acme/GreetingResource.java)
JCClassDecl(name = GreetingResource, line = 10)
JCClassDecl(name = HelloService, line = 22)
--- END PATH ---

[WARNING] Failed to convert for the following cursor stack:--- BEGIN PATH ---
JCCompilationUnit(sourceFile = /Users/manderse/temp/quarkus-timed-producers/src/main/java/org/acme/GreetingResource.java)
JCClassDecl(name = GreetingResource, line = 10)
--- END PATH ---

[WARNING] Failed to convert for the following cursor stack:--- BEGIN PATH ---
JCCompilationUnit(sourceFile = /Users/manderse/temp/quarkus-timed-producers/src/main/java/org/acme/GreetingResource.java)
--- END PATH ---

[WARNING] Unexpected modifier sealed
```

Despite the failure to parse the maven plugin reports all went fine - but it did not :)

The code that "breaks" openrewrite is:

```
public static class HelloService {
        public static String HELLO = "hello ";
        record Msg(String msg) {}
        sealed interface Foo permits Bar,Baz{}
        static final class Bar implements Foo {}
        static final class Baz implements Foo {}
        public String hello() {
            return new Msg(HELLO).toString() + new Bar();
        }
    }
```

I run it with Java 17 which I assume has a javac/java parser that supports `sealed` thus I'm assuming its somehow openrewrite internal model that is failing to capture sealed classes/interfaces?

**Comment by timtebeek** [comment]
Hi @maxandersen ! What version of OpenRewrite are your running? We [released 7.35.0 last week](https://github.com/openrewrite/rewrite/releases/tag/v7.35.0), which includes a [fix for sealed classes](https://github.com/openrewrite/rewrite/commit/6491026c8e85513e1b13be7ac726f65412e9a502).

**Comment by maxandersen** [comment]
doh, I should have checked with latest version. `rewrite-maven-plugin:4.39.0` does not give this error.

still diff is not being generated but at least it seems it is no longer related to seal classes. I'll investigate further.
